lightning/backend/local.go

```
@@ -1510,3 +1492,180 @@ func (s *sizeProperties) iter(f func(p *rangeProperty) bool) {
+	var writer *sstable.Writer = nil
+	//var wb *pebble.Batch = nil
+	var wb []common.KvPair
+	var filePath string
+	defer w.consumeWg.Done()
+	var err error
+	for kvs := range w.kvsChan {
+		atomic.AddInt64(&w.local.Length, int64(len(kvs)))
+		if len(wb) == 0 && w.isSorted(kvs) {
+			if writer == nil {
+				writer, filePath, err = w.createWriter()
+				if err != nil {
+					w.writeErr.Set(err)
+					return
+				}
+			}
+			sz, err := writeKVs(writer, kvs)
+			if err != nil {
+				w.writeErr.Set(err)
+				return
+			}
+			totalSize += sz
+		} else {
+			for _, pair := range kvs {
+				batchSize += int64(len(pair.Key) + len(pair.Val))
+			}
+			// If this is the first kv array reach, we can use it directly to reduce memory copy.
+			if len(wb) == 0 && totalSize == 0 {
+				wb = kvs
+			} else {
+				wb = append(wb, kvs...)
+			}
+
+			if batchSize > w.memtableSizeLimit {
+				if err := w.flushKVs(wb); err != nil {
+					w.writeErr.Set(err)
+					return
+				}
+				wb = wb[:0]
+				totalSize += batchSize
+				batchSize = 0
+			}
+		}
+	}
+
+	if wb != nil {
+		if batchSize > 0 {
+			if err := w.flushKVs(wb); err != nil {
+				w.writeErr.Set(err)
+				return
+			}
+			totalSize += batchSize
+		}
+		log.L().Info("write data by sort index", zap.Int64("bytes", totalSize))
+	}
+	if writer != nil {
+		if err := writer.Close(); err != nil {
+			w.writeErr.Set(err)
+			return
+		}
+		if err := w.local.db.Ingest([]string{filePath}); err != nil {
+			w.writeErr.Set(err)
+			return
+		}
+		log.L().Info("write data by sst writer", zap.Int64("bytes", totalSize))
+	}
+	atomic.AddInt64(&w.local.TotalSize, totalSize)
+}
+
+func (w *LocalWriter) flushKVs(kvs []common.KvPair) error {
+	writer, filePath, err := w.createWriter()
+	if err != nil {
+		return err
+	}
+	sort.Slice(kvs, func(i, j int) bool {
+		return bytes.Compare(kvs[i].Key, kvs[j].Key) < 0
+	})
+	if _, err := writeKVs(writer, kvs); err != nil {
+		return err
+	}
+	if err := writer.Close(); err != nil {
+		return nil
```

why ignores error here?

---

I have fixed it.